build/common.ps1

```diff
@@ -492,10 +573,23 @@ Function Invoke-ILMerge {
         [string]$Configuration = $DefaultConfiguration
     )
     $nugetArtifactsFolder = Join-Path $Artifacts "$Configuration\NuGet.Clients\NuGet.CommandLine"
-    pushd $nugetArtifactsFolder
+    $dlls = Get-ChildItem $nugetArtifactsFolder -Filter 'nuget.*.dll' | %{ $_.Name }
```

Comment:
here is a suggestion:

1. Create a whitelist and blacklist, they can be two files or one, they can be json or two simple flat files with each dll name on a line
2. When processing make sure each dll in the whitelist exists, or else fail with an error
3. make sure each dll exists either in the whitelist or the blacklist or else fail with an error
4. make sure each dll in the black list exists or warn if it is missing
5. Errors and warnings should point to the files to be edited explicitly
6. The CI process should use exactly the same files as the build script
7. This change should also apply to the vsix include/exclude process